pty/ptytest/ptytest.go

```diff
@@ -145,16 +146,36 @@ type outExpecter struct {
 }
 
 func (e *outExpecter) ExpectMatch(str string) string {
+	return e.expectMatchContextFunc(str, e.ExpectMatchContext)
+}
+
+func (e *outExpecter) ExpectRegexMatch(str string) string {
+	return e.expectMatchContextFunc(str, e.ExpectRegexMatchContext)
+}
+
+func (e *outExpecter) expectMatchContextFunc(str string, impl func(ctx context.Context, str string) string) string {
 	e.t.Helper()
 
 	timeout, cancel := context.WithTimeout(context.Background(), testutil.WaitMedium)
 	defer cancel()
 
-	return e.ExpectMatchContext(timeout, str)
+	return impl(timeout, str)
 }
 
 // TODO(mafredri): Rename this to ExpectMatch when refactoring.
```

Comment:
@mafredri not really sure what this TODO is about since both this and the `ExpectMatch` functions have lots of usage. Could you elaborate please?

Reply:
Essentially this should be marked deprecated. All usage should be switched to use the context variant. Then once that's done. Delete this one and rename the context variant to this.

Reply:
OK, is it because we need to be able to control the deadline?
I think I'll do that in a separate PR.